roles/weaviate/defaults/main.yml

```diff
@@ -0,0 +1,120 @@
+# Store this in Ansible Vault for production deployments
+# Example: "dGhpcyBpcyBhIHNlY3VyZSBhcGkga2V5IGZvciBkZW1vIHB1cnBvc2VzIG9ubHk="
+weaviate_api_key: ""
+
+# Username associated with the API key
+weaviate_api_user: "admin"
+
+# Logging
+# Options: trace, debug, info, warning, error, fatal, panic
+weaviate_log_level: "info"
+
+# Let's Encrypt Configuration (REQUIRED - must be set in playbook or inventory)
+# Email address for Let's Encrypt certificate notifications and renewal reminders
+# Example: "admin@example.com"
+letsencrypt_email: ""
+
+# IP Safelisting (Optional)
+# Default: accessible from everywhere with API key authentication
+# Examples:
+#   Single IP: "203.0.113.1/32"
+#   Multiple: "10.0.0.0/8,192.168.0.0/16,172.16.0.0/12"
+#   Private networks: "10.0.0.0/8,192.168.0.0/16,172.16.0.0/12"
+weaviate_allowed_ips: ""
+
+##############################################################################
+# Multi2vec-CLIP Configuration
+##############################################################################
+
+weaviate_multi2vec_image: "cr.weaviate.io/semitechnologies/multi2vec-clip:sentence-transformers-clip-ViT-B-32-multilingual-v1"
+weaviate_enable_cuda: "0"
+
+##############################################################################
+# Resource Limits
+##############################################################################
+#
+# IMPORTANT: The defaults below are optimized for DEVELOPMENT/TESTING on
+# modest hardware (8-12GB RAM total). For PRODUCTION deployments, you should
+# increase these limits based on your workload and available resources.
+#
+# Default totals (development):
+#   - CPU Limits: 4 cores, Reservations: 2 cores
+#   - Memory Limits: 6.5GB, Reservations: 3.25GB
+#
+# Recommended production values:
+#   - Small (16GB RAM): Weaviate 4CPU/8GB, CLIP 2CPU/4GB, Traefik 1CPU/1GB
+#   - Medium (32GB RAM): Weaviate 6CPU/12GB, CLIP 3CPU/6GB, Traefik 1CPU/1GB
+#   - Large (64GB+ RAM): Weaviate 8CPU/16GB, CLIP 4CPU/8GB, Traefik 1CPU/1GB
+#
+# See README.md for detailed resource planning guidance.
+##############################################################################
+
+# Traefik Resources (reverse proxy - minimal requirements)
+# Note: Traefik is very lightweight; 1 CPU is more than sufficient
+weaviate_traefik_cpu_limit: "1"
+weaviate_traefik_memory_limit: "512M"
+weaviate_traefik_cpu_reservation: "1"
+weaviate_traefik_memory_reservation: "256M"
+
+# Weaviate Resources (vector database - adjust based on dataset size)
+# Development defaults support small datasets and light usage
+weaviate_cpu_limit: "2"
+weaviate_memory_limit: "4G"
+weaviate_cpu_reservation: "1"
+weaviate_memory_reservation: "2G"
+
+# Multi2vec-CLIP Resources (ML model - CPU-only inference)
+# Development defaults for basic embedding generation
+weaviate_multi2vec_cpu_limit: "1"
+weaviate_multi2vec_memory_limit: "2G"
+weaviate_multi2vec_cpu_reservation: "1"
+weaviate_multi2vec_memory_reservation: "1G"
```

Comment:
I’d recommend keeping memory_limit and memory_reservation the same for these services. If the reservation is lower than the limit, a scheduler could pack more tasks onto a node than the actual peak memory usage allows, which can lead to overcommit - at least that's what I remember from using k8s